test/MVCFramework45.FunctionalTests/Controllers/HomeController.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights;
+
+namespace MVCFramework45.FunctionalTests.Controllers
+{
+    public class HomeController : Controller
+    {
+        private TelemetryClient telemetryClient;
+
+        public HomeController(TelemetryClient telemetryClient)
+        {
+            this.telemetryClient = telemetryClient;
+        }
+
+        //private IApplicationEnvironment applicationEnvironment;
+
+        //public HomeController(TelemetryClient telemetryClient, IApplicationEnvironment applicationEnvironment)
+        //{
+        //    this.telemetryClient = telemetryClient;
+        //    this.applicationEnvironment = applicationEnvironment;
+        //}
+
+        public IActionResult Index()
+        {
+            return View();
+        }
+
+        public IActionResult Exception()
+        {
+            throw new InvalidOperationException("Do not call the method called Exception");
+        }
+
+        public IActionResult About(int index)
+        {
+            ViewBag.Message = "Your application description page # " + index;
+
+            return View();
+        }
+
+        public async Task<IActionResult> Contact()
+        {
+            this.telemetryClient.TrackEvent("GetContact");
+            this.telemetryClient.TrackMetric("ContactFile", 1);
+            this.telemetryClient.TrackTrace("Fetched contact details.", SeverityLevel.Information);
+            return View();
+        }
+
+        public IActionResult Error()
+        {
+            return View("~/Views/Shared/Error.cshtml");
+        }
+
+        //private async Task<string> GetContactDetails()
+        //{
```

Comment:
@davidfowl @Eilon We are seeing some exceptions when we uncomment this section of code. We are getting the following exception:
`System.InvalidOperationException: Unable to resolve service for type 'Microsoft.Extensions.CodeGeneration.DotNet.IApplicationEnvironment' while attempting to activate 'MVCFramework45.FunctionalTests.Controllers.HomeController'.
   at Microsoft.Extensions.Internal.ActivatorUtilities.GetService(IServiceProvider sp, Type type, Type requiredBy, Boolean isDefaultParameterRequired)`

Please let me know if this is something we need to configure. We actually get it from dependency injection (commented out section on top).